### PR TITLE
Add "Closes" comment to PR with failing CI

### PR DESCRIPTION
**Pull request by dhaiducek**

### CODE REVIEW

**Review by JustinKuli — COMMENTED** [review]
I like this idea (and the code looks good), but I don't know if a comment actually does this. I tried that in https://github.com/stolostron/governance-policy-propagator/pull/386, but it didn't automatically close. 

https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue suggests it only works on *PR descriptions*

**Comment by dhaiducek** [comment]
Updated. I've left the comment function in there since in the future I'd like to implement some slash commands (like `/auto-cc` or `/assign`)

File: src/web.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { ApplicationFunctionOptions, Probot, Server } from "probot";
 
 import { Config, loadConfig } from "./config";
 import { Database, PRAction } from "./db";
-import { createFailureIssue, getRequiredChecks, mergePR } from "./github";
+import { addComment, appendPRDescription, createFailureIssue, getRequiredChecks, mergePR } from "./github";
```
Comment (JustinKuli) [inline src/web.ts:7]: Nit: does `addComment` still need to be imported? I guess I'm surprised a linter or something didn't get upset at this.

**Review by JustinKuli — APPROVED** [review]
Optional nits. Otherwise looks great!

/hold

File: src/web.ts (hunk 1 above)
Comment (dhaiducek) [reply src/web.ts:7]: Good catch. I noticed packages get added automatically but not removed. I might be missing some tooling around TypeScript...

**Review by JustinKuli — APPROVED** [review]

**Comment by JustinKuli** [comment]
Feel free to unhold

File: src/github.ts [hunk 2]
```diff
@@ -56,6 +56,65 @@ export async function createFailureIssue(
...
+/**
+ * Comment on a GitHub issue (or pull request).
+ * @param {Octokit} client the GitHub client to use that is authenticated as the GitHub installation.
+ * @param {string} org the GitHub organization where the issue is to add a comment.
+ * @param {string} repo the GitHub repository where the issue is to add a comment.
+ * @param {number} id the issue ID.
+ * @param {string} message the content of the comment.
+ * @return {Promise<boolean>}
+ */
+export async function addComment(
```
Comment (mprahl) [inline src/github.ts:68]: @dhaiducek could you please remove this unused function?

File: src/web.ts [hunk 3]
```diff
@@ -263,6 +263,20 @@ export async function app(probot: Probot, probotOptions: ApplicationFunctionOpti
       pendingPR.action = PRAction.Blocked;
       await db.setPendingPR(pendingPR);
 
+      // Append to description on PR to attach the issue to the pending PR
+      const closesMsg = `Closes #${issueID}`;
+      const appended = await appendPRDescription(
```
Comment (mprahl) [inline src/web.ts:268]: I think a cleaner approach would have been to still throw an error if the API call fails but let the caller (the code here) choose to log and ignore it. The reason of the error is lost to the caller in this case.

File: src/github.ts (hunk 2 above)
Comment (dhaiducek) [reply src/github.ts:68]: I plan to use it in the future, but I can remove it if that's preferred:
https://github.com/stolostron/magic-mirror/pull/36#issuecomment-1553693440

File: src/web.ts (hunk 3 above)
Comment (dhaiducek) [reply src/web.ts:268]: Good point--I'll make an update

**Pull request merged**